src/elements/paginator/paginator.ts

```diff
@@ -0,0 +1,344 @@
+import {
+  type CSSResultGroup,
+  html,
+  LitElement,
+  nothing,
+  type PropertyValues,
+  type TemplateResult,
+} from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { repeat } from 'lit/directives/repeat.js';
+
+import { getNextElementIndex, isArrowKeyPressed, sbbInputModalityDetector } from '../core/a11y.js';
+import { SbbConnectedAbortController, SbbLanguageController } from '../core/controllers.js';
+import { EventEmitter } from '../core/eventing.js';
+import { i18nPreviousPage, i18nNextPage, i18nPage } from '../core/i18n.js';
+import { SbbNegativeMixin } from '../core/mixins.js';
+import type { SbbSelectElement } from '../select.js';
+
+import style from './paginator.scss?lit&inline';
+
+import '../button/mini-button.js';
+import '../button/mini-button-group.js';
+import '../divider.js';
+import '../form-field.js';
+import '../select.js';
+import '../option.js';
+
+export type SbbPaginatorPageChanged = {
+  currentPageIndex: number;
+  previousPageIndex: number;
+};
+
+const MAX_PAGE_NUMBERS_DISPLAYED = 3;
+
+/**
+ * It displays a paginator component.
+ *
+ * @event {CustomEvent<SbbPaginatorPageChanged>} pageChanged - Emits when the pageIndex changes.
+ */
+@customElement('sbb-paginator')
+export class SbbPaginatorElement extends SbbNegativeMixin(LitElement) {
+  public static override styles: CSSResultGroup = style;
+  public static readonly events: Record<string, string> = {
+    pageChanged: 'pageChanged',
+  } as const;
+
+  /** Total number of items. */
+  @property({ type: Number })
+  public set length(value: number) {
+    this._length = isNaN(value) ? 0 : Math.max(value, 0);
+  }
+  public get length(): number {
+    return this._length;
+  }
+  private _length: number = 0;
+
+  /** Number of items per page. */
+  @property({ attribute: 'page-size', type: Number })
+  public set pageSize(value: number) {
+    this._pageSize = Math.max(value, 0);
+    this.pageIndex = Math.floor((this.pageIndex * this.pageSize) / this._pageSize) || 0;
```

Comment:
Not sure what you meant to do with this 🤔 
I believe it does nothing since you're multiplying and dividing for `pageSize`

There's a case that is not handled and for which we should decide how to behave. 
Suppose you are on the last page and the user increases the page size. Your page does not exist anymore, I think you should move to the new last index

Reply:
Fixed in [3a8d216](https://github.com/sbb-design-systems/lyne-components/pull/2982/commits/3a8d216e75cc1f939e0499823fe2bab7f2078c86)